dGame/dComponents/RocketLaunchLupComponent.cpp

```diff
@@ -21,20 +21,17 @@ RocketLaunchLupComponent::RocketLaunchLupComponent(Entity* parent) : Component(p
 RocketLaunchLupComponent::~RocketLaunchLupComponent() {}
 
 void RocketLaunchLupComponent::OnUse(Entity* originator) {
-	// the LUP world menu is just the property menu, the client knows how to handle it
-	GameMessages::SendPropertyEntranceBegin(m_Parent->GetObjectID(), m_Parent->GetSystemAddress());
-
-	// get the rocket to "equip" it so we are holding it
-	// taken from the RocketLaunchControlComponent
-	auto* inventoryComponent = originator->GetComponent<InventoryComponent>();
 	auto* characterComponent = originator->GetComponent<CharacterComponent>();
+	auto* inventoryComponent = originator->GetComponent<InventoryComponent>();
 
-	if (!inventoryComponent || !characterComponent) return;
+	if (!characterComponent || !inventoryComponent) return;
 
-	Item* rocket = inventoryComponent->FindItemById(characterComponent->GetLastRocketItemID());
+	// Make sure we have a valid rocket
+	auto* rocket = inventoryComponent->FindItemById(characterComponent->GetLastRocketItemID());
```

Comment:
Same here as above - What will happen here if the player has no previously used rocket item id?

Reply:
Isnt this also supposed to equip the rocket as it does in live?  It looks like rocket isnt even used here

Reply:
the equip is handle in RocketLaunchpadComponent::OnUse, but I think the better way to do it is each component when it's actually needed, instead of doing multiple checks